### PR TITLE
pyo3 0.5 support

### PR DESCRIPTION
**Pull request by mre**
This updates pyo3 to 0.5.
The changelog can be found [here](https://github.com/PyO3/pyo3/blob/master/CHANGELOG.md#050---2018-11-11).
Special thanks to @konstin, who helped me a lot to get this done. Couldn't make it work without him.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,16 +10,16 @@ serde = "1.0.53"
 serde_derive = "1.0.53"
 
 [dependencies.pyo3]
-version = "0.4.0"
+version = "0.5.0"
 features = ["extension-module"]
```
Comment (konstin) [inline Cargo.toml]: That line shouldn't be here

File: src/lib.rs [hunk 2]
```diff
@@ -293,8 +288,8 @@ struct SerializePyObject<'p, 'a> {
 
 impl<'p, 'a> Serialize for SerializePyObject<'p, 'a> {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
+        where
```
Comment (konstin) [inline src/lib.rs]: That's what intellij-rust's default formatter does, but unfortunately that's not rustfmt style. You can use rustfmt though through he "format with rustfmt" action. (See [intellij-rust/intellij-rust#431](https://github.com/intellij-rust/intellij-rust/issues/431))

File: Cargo.toml (hunk 1 above)
Comment (mre) [reply Cargo.toml]: This happens when I code at night and cook noodles on the side:
I forget to commit half of my changes while in the meantime the kitchen almost burns down. :man_cook: :disappointed:

File: src/lib.rs (hunk 2 above)
Comment (mre) [reply src/lib.rs]: Glad that at least one of us reads my code. :laughing: Found the option. Rustfmt all the things.

File: Cargo.toml (hunk 1 above)
Comment (konstin) [reply Cargo.toml]: And they told us doing things in parallel would be fine in rust :wink:  🍜

Comment (mre) [reply Cargo.toml]: BTW tweeted that: https://twitter.com/matthiasendler/status/1062838581914427393

**Pull request merged**